packages/router/src/router.tsx

```diff
@@ -144,29 +146,35 @@ const InternalRoute: React.VFC<InternalRouteProps> = ({
   )
 }
 
-interface PrivateProps {
+type PrivateProps<P> = Omit<SetProps<P>, 'wrap'> & {
   /** The page name where a user will be redirected when not authenticated */
   unauthenticated: string
   role?: string | string[]
+  wrap?: WrapperType<P> | WrapperType<P>[]
 }
 
 /**
  * `Routes` nested in `Private` require authentication.
  * When a user is not authenticated and attempts to visit this route they will be
  * redirected to `unauthenticated` route.
  */
-const Private: React.FC<PrivateProps> = ({
-  children,
-  unauthenticated,
-  role,
-}) => {
+function Private<WrapperProps>(props: PrivateProps<WrapperProps>) {
+  const { role, unauthenticated, children, wrap, ...rest } = props
```

Comment:
What could `...rest` include here?

Reply:
`<Set>`s allow you to pass arbitrary props to the wrapper components, like `<Set wrap={MainLayout} theme='dark'>` will pass `theme='dark'` to `MainLayout`. I wanted to have the same behavior for `<Private>`. So that's what `...rest` will include

Reply:
Oooh, I didn't realize that was a thing! What a great feature. I guess I should read the docs :)